### PR TITLE
[Merl 910] Add CoreButton and CoreButtons blocks

### PR DESCRIPTION
**Pull request by theodesp**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.

## Description

* Provides reference implementation of CoreButton and CoreButtons block.

## Testing

Follow instructions to configure the blocks package in the example project. https://faustjs.org/docs/gutenberg/getting-started

* Generate the global stylessheet and import it into the project. faust generateGlobalStylesheet.
```
import { WordPressBlocksProvider, fromThemeJson } from '@faustwp/blocks';
import '../globalStylesheet.css';

export default function MyApp({ Component, pageProps }) {
  const router = useRouter();
  return (
    <FaustProvider pageProps={pageProps}>
      <WordPressBlocksProvider
        config={{
          blocks,
          theme: {},
        }}>
        <Component {...pageProps} key={router.asPath} />
      </WordPressBlocksProvider>
    </FaustProvider>
  );
}
```
Comment out all the styles in the examples/next/faustwp-getting-started/styles/_blocks.scss since they will interfere with the global stylesheet.
```
Import the blocks:
import { CoreBlocks } from '@faustwp/blocks';

export default {
  ...CoreBlocks,
};
```
Make sure you use the block fragment in the page query:
```
import components from '../wp-blocks';
import { flatListToHierarchical } from '@faustwp/core';
import { WordPressBlocksViewer } from '@faustwp/blocks';
...
const { title, content, featuredImage, date, author, editorBlocks } = props.data.post;
const blocks = flatListToHierarchical(editorBlocks, {childrenKey: 'innerBlocks'});
${components.CoreButton.fragments.entry}
${components.CoreButtons.fragments.entry}
...
editorBlocks {
        name
        __typename
        renderedHtml
        id: clientId
        parentId: parentClientId
        ...${components.CoreButton.fragments.key}
        ...${components.CoreButtons.fragments.key}
      }
```
Use the WordPressBlocksViewer:
```
<Container className="wp-block-group is-layout-flow">
            <ContentWrapper className="entry-content wp-block-post-content has-global-padding is-layout-constrained">
              <WordPressBlocksViewer blocks={blocks}/>
            </ContentWrapper>
          </Container>
```
Go to WordPress site and create several buttons combining styles, links and widths All changes should reflect in the decoupled site when updating the post.

## Screenshots

Screenshot 2023-06-14 at 13 18 34

### CODE REVIEW

**Comment by mindctrl** [comment]
@theodesp thanks for the extra example info. Does it work for you as is in this PR in a clean dir? It seems like something is missing.

**Comment by theodesp** [comment]
> @theodesp thanks for the extra example info. Does it work for you as is in this PR in a clean dir? It seems like something is missing.

Yes. It only renders a page with `Buttons` and `Button` components ATM. Also the globalStylesheet that is provided is using the `TwentyTwentyThree theme`.

**Comment by blakewilson** [comment]
@theodesp I'm trying to test this but I can't seem to render any of the buttons.

Here are a few screenshots:

Screenshot 2023-06-20 at 1 37 35 PM

Screenshot 2023-06-20 at 1 37 17 PM

Screenshot 2023-06-20 at 1 37 04 PM

You can see I'm getting the blocks back in the `console.log` statement just fine. I think maybe it has something to do with applying the proper `innerBlocks`?

**Comment by theodesp** [comment]
@blakewilson there is a testing commit here
https://github.com/wpengine/faustjs/pull/1461/commits/e71c34a0d6e6c10ef6f60963d254a75d7a7dcffc

I will update the testing instructions, the flatListToHierarchical needs to have the proper `childrenKey`.

**Comment by mindctrl** [comment]
> there is a testing commit here https://github.com/wpengine/faustjs/commit/e71c34a0d6e6c10ef6f60963d254a75d7a7dcffc

This worked for me. Steps I had to take:
- `npm i` in repo root
- `npm run build` in repo root
- `cd examples/next/faustwp-getting-started`
- `npm run generate && npm run stylesheet`
- `npm run dev`

**Review by mindctrl — APPROVED** [review]
LGTM, but would like @blakewilson to confirm too before merging.

File: packages/blocks/package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "react": ">=17.0.2",
     "react-dom": ">=17.0.2",
     "@wordpress/style-engine": ">=1.18.0",
-    "@apollo/client": ">=3.6.6"
+    "@apollo/client": ">=3.6.6",
+    "next": ">=12.1.6"
```
Comment (blakewilson) [inline packages/blocks/package.json:14]: We may want this to be a peer dep.

Comment (blakewilson) [inline packages/blocks/package.json:14]: Would adding this new peer dep be a breaking change? @theodesp @mindctrl

**Pull request merged**